packages/react-core/src/hooks/useUserAttributes/action.ts

```diff
@@ -0,0 +1,172 @@
+import {
+  confirmUserAttribute,
+  deleteUserAttributes,
+  fetchUserAttributes,
+  sendUserAttributeVerificationCode,
+  updateUserAttributes,
+  UserAttributeKey,
+} from '@aws-amplify/auth';
+
+import {
+  ConfirmInput,
+  DeleteInput,
+  FetchInput,
+  HandleAttributeActionInput,
+  HandleAttributeActionOutput,
+  SendCodeInput,
+  UpdateInput,
+} from './types';
+
+const confirmAction = async (
+  prevResult: HandleAttributeActionOutput,
+  { userAttributeKey, confirmationCode }: ConfirmInput
+): Promise<HandleAttributeActionOutput> => {
+  await confirmUserAttribute({ userAttributeKey, confirmationCode });
+  const attributes = await fetchUserAttributes();
+  return {
+    ...prevResult,
+    attributes,
+    pendingVerification: [{ name: userAttributeKey }],
+  };
+};
+
+const deleteAction = async (
+  prevResult: HandleAttributeActionOutput,
+  { userAttributeKeys }: DeleteInput
+): Promise<HandleAttributeActionOutput> => {
+  // cast due to underlying API interface forcing a non-idiomatic
+  // structure of `userAttributeKeys`
+  await deleteUserAttributes({ userAttributeKeys } as {
+    userAttributeKeys: [UserAttributeKey, ...UserAttributeKey[]];
+  });
+  const attributes = await fetchUserAttributes();
+  return { ...prevResult, attributes };
+};
+
+const fetchAction = async (
+  prevResult: HandleAttributeActionOutput,
+  _: FetchInput
+): Promise<HandleAttributeActionOutput> => {
+  const attributes = await fetchUserAttributes();
+  return { ...prevResult, attributes };
+};
+
+const sendCodeAction = async (
+  prevResult: HandleAttributeActionOutput,
+  { userAttributeKey }: SendCodeInput
+): Promise<HandleAttributeActionOutput> => {
+  const result = await sendUserAttributeVerificationCode({ userAttributeKey });
+  return {
+    ...prevResult,
+    pendingVerification: [
+      {
+        name: userAttributeKey,
+        codeDeliveryDetails: {
+          destination: result.destination,
+          medium: result.deliveryMedium,
+        },
+      },
+    ],
```

Comment:
Will this remove values from `pendingVerification` that aren't part of the `sendCodeAction` input?